src/types/spectral.ts

```diff
@@ -53,7 +54,7 @@ export interface IResolver {
   resolve(source: unknown, opts?: IResolveOpts): Promise<ResolveResult>;
 }
 
-export type FormatLookup = (document: unknown) => boolean;
+export type FormatLookup = (document: unknown, source?: string) => boolean;
```

Comment:
```suggestion
export type FormatLookup = (document: unknown, source: string | null) => boolean;
```

Reply:
That would make it a breaking change, wouldn't it?

Reply:
Aye, that'd be indeed a breaking change.
Sounds good. Let's do it this way. Perhaps leave a comment for me so that I revisit it when we release a major version.

Reply:
```suggestion
// todo(@p0lip): make it string | null when working on 6.0
export type FormatLookup = (document: unknown, source?: string) => boolean;
```

Reply:
@P0lip Just out of curiosity, why favoring `null` when it can already be `undefined`?

Reply:
Cause document.source is either a null or string